### ISSUE TITLE
Failed to configure LogStashLogger

### ISSUE THREAD

**Issue by retgoat** [dwbutler/logstash-logger#36]
Hi @dwbutler,
I have some strange issue with LogStashLogger.
I try to config it to add some custom fields and my app fail to start with the next error:
```
/media/sf_offside/pspgateway/config/environments/development.rb:37:in `block in <top (required)>': undefined method `configure' for LogStashLogger:Module (NoMethodError)
	from /home/retgoat/.rvm/gems/ruby-2.1.2/gems/railties-3.2.13/lib/rails/railtie/configurable.rb:24:in `class_eval'
	from /home/retgoat/.rvm/gems/ruby-2.1.2/gems/railties-3.2.13/lib/rails/railtie/configurable.rb:24:in `configure'
	from /media/sf_offside/pspgateway/config/environments/development.rb:2:in `<top (required)>'
	from /home/retgoat/.rvm/gems/ruby-2.1.2/gems/activesupport-3.2.13/lib/active_support/dependencies.rb:251:in `require'
	from /home/retgoat/.rvm/gems/ruby-2.1.2/gems/activesupport-3.2.13/lib/active_support/dependencies.rb:251:in `block in require'
	from /home/retgoat/.rvm/gems/ruby-2.1.2/gems/activesupport-3.2.13/lib/active_support/dependencies.rb:236:in `load_dependency'
	from /home/retgoat/.rvm/gems/ruby-2.1.2/gems/activesupport-3.2.13/lib/active_support/dependencies.rb:251:in `require'
	from /home/retgoat/.rvm/gems/ruby-2.1.2/gems/railties-3.2.13/lib/rails/engine.rb:571:in `block in <class:Engine>'
	from /home/retgoat/.rvm/gems/ruby-2.1.2/gems/railties-3.2.13/lib/rails/initializable.rb:30:in `instance_exec'
	from /home/retgoat/.rvm/gems/ruby-2.1.2/gems/railties-3.2.13/lib/rails/initializable.rb:30:in `run'
	from /home/retgoat/.rvm/gems/ruby-2.1.2/gems/railties-3.2.13/lib/rails/initializable.rb:55:in `block in run_initializers'
	from /home/retgoat/.rvm/gems/ruby-2.1.2/gems/railties-3.2.13/lib/rails/initializable.rb:54:in `each'
	from /home/retgoat/.rvm/gems/ruby-2.1.2/gems/railties-3.2.13/lib/rails/initializable.rb:54:in `run_initializers'
```

Here is my Gemfile
```ruby
gem 'logstash-logger', '~>0.8.0'
```

Here is the initialization
```ruby
#config/environments/development.rb
require 'logstash-logger'

#some code
config.logger = LogStashLogger.new([{type: :file, path: 'log/development.log'}, {type: :udp, host: 'localhost', port: 5923}])
  LogStashLogger.configure do |config|
    config.customize_event do |event|
      event["@caller"] = caller[6] ? caller[6].split("/").last : "UNDEFINED"
    end
  end

#some code
```

But it I try to use cloned logstash-logger like this:

```ruby
#Gemfile
gem 'logstash-logger', :path => '../logstash-logger'
```
All working like a charm!

Rails 3.2.13
Ruby 2.1.2

Could you please help with this?
Thanks!

**Comment by stewartmckee** [comment]
Just had the same issue.... Looks like its just new and is not in the latest bundled gem...  its on the master branch but if you switch to the tag for 0.8.0 it isn't mentioned in the readme... Correct me if i'm wrong though.

If you really need it you could change your Gemfile to be 

    gem 'logstash-logger', :git => 'git@github.com:dwbutler/logstash-logger.git'

that will essentially do the same as you did by cloning the repository.

**Comment by retgoat** [comment]
Hi @stewartmckee,
yep, I know if I'll plug this gem from github it will work correctly. But because of company's policy I can't do that.
Anyway, many thanks for your comment :) That means that I'm not alone :D

**Comment by retgoat** [comment]
It seems I had found an issue.
Here is the directory listing of **cloned** gem:
```
├── Appraisals
├── CHANGELOG.md
├── Gemfile
├── gemfiles
│   ├── rails_3.2.gemfile
│   ├── rails_4.0.gemfile
│   ├── rails_4.1.gemfile
│   └── rails_4.2.gemfile
├── lib
│   ├── logstash-logger
│   │   ├── configuration.rb
│   │   ├── device
│   │   │   ├── base.rb
│   │   │   ├── connectable.rb
│   │   │   ├── file.rb
│   │   │   ├── io.rb
│   │   │   ├── multi_delegator.rb
│   │   │   ├── redis.rb
│   │   │   ├── socket.rb
│   │   │   ├── stderr.rb
│   │   │   ├── stdout.rb
│   │   │   ├── tcp.rb
│   │   │   ├── udp.rb
│   │   │   └── unix.rb
│   │   ├── device.rb
│   │   ├── formatter.rb
│   │   ├── logger.rb
│   │   ├── railtie.rb
│   │   ├── tagged_logging.rb
│   │   └── version.rb
│   └── logstash-logger.rb
├── LICENSE.txt
├── logstash-logger.gemspec
├── logstash-logger.sublime-project
├── logstash-logger.sublime-workspace
├── Rakefile
├── README.md
├── samples
│   ├── example.crt
│   ├── example.key
│   ├── file.conf
│   ├── redis.conf
│   ├── ssl.conf
│   ├── tcp.conf
│   ├── udp.conf
│   └── unix.conf
└── spec
    ├── configuration_spec.rb
    ├── device
    │   ├── file_spec.rb
    │   ├── io_spec.rb
    │   ├── multi_delegator_spec.rb
    │   ├── redis_spec.rb
    │   ├── socket_spec.rb
    │   ├── stderr_spec.rb
    │   ├── stdout_spec.rb
    │   ├── tcp_spec.rb
    │   ├── udp_spec.rb
    │   └── unix_spec.rb
    ├── device_spec.rb
    ├── logger_spec.rb
    ├── rails_spec.rb
    ├── spec_helper.rb
    └── tagged_logging_spec.rb
```

And this is listing of gem downloaded from rubygems:

```
├── Appraisals
├── CHANGELOG.md
├── Gemfile
├── gemfiles
│   ├── rails_3.2.gemfile
│   ├── rails_4.0.gemfile
│   ├── rails_4.1.gemfile
│   └── rails_4.2.gemfile
├── lib
│   ├── logstash-logger
│   │   ├── device
│   │   │   ├── base.rb
│   │   │   ├── connectable.rb
│   │   │   ├── file.rb
│   │   │   ├── io.rb
│   │   │   ├── multi_delegator.rb
│   │   │   ├── redis.rb
│   │   │   ├── socket.rb
│   │   │   ├── stderr.rb
│   │   │   ├── stdout.rb
│   │   │   ├── tcp.rb
│   │   │   ├── udp.rb
│   │   │   └── unix.rb
│   │   ├── device.rb
│   │   ├── formatter.rb
│   │   ├── logger.rb
│   │   ├── railtie.rb
│   │   ├── tagged_logging.rb
│   │   └── version.rb
│   └── logstash-logger.rb
├── LICENSE.txt
├── logstash-logger.gemspec
├── Rakefile
├── README.md
├── samples
│   ├── example.crt
│   ├── example.key
│   ├── file.conf
│   ├── redis.conf
│   ├── ssl.conf
│   ├── tcp.conf
│   ├── udp.conf
│   └── unix.conf
└── spec
    ├── device
    │   ├── file_spec.rb
    │   ├── io_spec.rb
    │   ├── multi_delegator_spec.rb
    │   ├── redis_spec.rb
    │   ├── socket_spec.rb
    │   ├── stderr_spec.rb
    │   ├── stdout_spec.rb
    │   ├── tcp_spec.rb
    │   ├── udp_spec.rb
    │   └── unix_spec.rb
    ├── device_spec.rb
    ├── logger_spec.rb
    ├── rails_spec.rb
    ├── spec_helper.rb
    └── tagged_logging_spec.rb
```

There is a lack of configuration.rb in the second listing.
@dwbutler need to update gem in rubygems, I think.

**Comment by retgoat** [comment]
BTW, version 0.7.0 had the same issue too.

**Comment by dwbutler** [comment]
The configuration feature is not released yet. I think it's stable enough to release now.

For future reference, you can check the project changelog. This file mentions that the config feature is Edge rather than released.

**Comment by retgoat** [comment]
@dwbutler, many thanks.

**Comment by dwbutler** [comment]
This has been released as 0.9.0.
